File: sync/src/main/java/com/inceptionnotes/sync/events/ShowEvent.java
package com.inceptionnotes.sync.events;

import com.inceptionnotes.sync.Client;

/**
 * Created by jacob on 2/10/18.
 */

public class ShowEvent extends Event {

    public String show;

    @Override
    public void got(Client client) {
        if (!client.isIdentified()) {
            client.send(new SimpleMessageServerEvent("hey matey. plz identify urself matey. thx matey."));
            return;
        }

        client.setShow(show);
    }
}

File: sync/src/main/java/com/inceptionnotes/sync/events/ShowEvent.kt
package com.inceptionnotes.sync.events

import com.inceptionnotes.sync.world.Client

/**
 * Created by jacob on 2/10/18.
 */

class ShowEvent : Event {

    var show: String? = null

    override fun got(client: Client) {
        if (!client.isIdentified) {
            client.send(SimpleMessageServerEvent("hey matey. plz identify urself matey. thx matey."))
            return
        }

        client.show = show
    }
}
